src/AbstractOperations/conditional_operations.jl

```diff
@@ -86,11 +89,22 @@ function ConditionalOperation(operand::AbstractField;
     return ConditionalOperation{LX, LY, LZ}(operand, func, operand.grid, condition, mask)
 end
 
+validate_condition(cond::Function, ::ConditionalOperation) = cond
+
+function validate_condition(cond::AbstractArray, c::ConditionalOperation)
+    if ndims(cond) != 3
+        throw(ArgumentError("when kwarg condition::Array requires a 3D array of size $(size(c))"))
+    end
+    return cond
+end
+
 function ConditionalOperation(c::ConditionalOperation;
                               func = c.func,
                               condition = c.condition,
                               mask = c.mask)
 
+    condition = validate_condition(condition, c)
```

Comment:
I'm confused how the input is `ConditionalOperation`. Possibly this is cleaned up in #3794

Reply:
hm.... yes, I got it from there....